### ISSUE TITLE
prefer-conditional-expression: check returns in addition to assignments

### ISSUE THREAD

**Issue by ajafff** [palantir/tslint#3136]
`prefer-conditional-expression` complains about this code:

```ts
let v: number;
if (condition) {
    v = 1;
} else {
    v = 2;
}
// prefer
const v = condition ? 1 : 2;
```

I'd like the rule to also check `return`. That means it should complain about this code as well:

```ts
if (condition) {
    return 1;
} else {
    return 2;
}
// prefer
return condition ? 1 : 2;
```

I don't see a difference to the existing behavior. That's why I think we don't need to add an option for this new check.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
